portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,14 +27,19 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments = new ArrayList<String>();
```

Comment:
Make this final to avoid unnecessary re-assignments.

Note that it is a list so you can still add/remove entries. `final` means you cannot assign `comments` to a new list.

Also, you can write

```
private final ArrayList<String> comments = new ArrayList<>();
```
Java8 can infer the type